src/proposal.mligo

```diff
@@ -344,7 +352,7 @@ let handle_proposal_is_over
     )
     : (operation list * storage * counter) =
   let proposal = check_if_proposal_exist (proposal_key, store) in
-  if    is_voting_period_over (proposal, config.voting_period, store)
+  if    is_proposal_flush_time_reached (proposal, config.proposal_flush_time)
```

Comment:
We need to now check for another condition here, which is that:
> proposals cannot be flushed after `proposal_expired_time` has passed from their submission

and fail when that happens, so:
```suggestion
  if is_proposal_expired_time_reached (proposal, config.proposal_expired_time)
  then (failwith("EXPIRED_PROPOSAL") : (operation list * storage * counter))
  else if is_proposal_flush_time_reached (proposal, config.proposal_flush_time)
```

Reply:
I see, but shouldn't we skip it instead? If it fails here, `flush` will keep failing if the amount of proposals to be `flush` include an expired proposal in it. 

Or we want to do it this way to keep the order of execution?

Reply:
Yep, we want to keep the order of execution and not allow expired proposals to be flushed.

`flush` will indeed fail if one (or more) proposals are expired, but that why we keep `drop_proposal` and someone will have to call it to move forward.

Reply:
I see, I've updated this now.